### PR TITLE
feat: add Outline sidebar panel with LSP document symbols

### PR DESCRIPTION
**Pull request by tenox7**
## Summary

Adds an **Outline** tab to the sidebar (after Changes) showing the symbol tree of the active file — functions, methods, types with fields, markdown sections — with click/enter to jump to the symbol.

```
│ Explore  Find  Outline  »  ⋮ ││ manager.go x │
│──────────────────────────────│┘              └──────────────────────
│▼ ◆ Manager                   │   1  package lsp
│   ▪ servers                  │   2
│   ▪ config                   │   3  import (
│   ▪ mu                       │   4    "fmt"
│   ▪ OnDiagnostics            │   5    "log/slog"
│ƒ NewManager                  │   6    "path/filepath"
│ƒ (*Manager).ServerConfig     │   7    "strings"
│ƒ (*Manager).ClientForLangu…  │   8    "sync"
```

- **LSP `textDocument/documentSymbol`** support in `internal/lsp`: handles both hierarchical `DocumentSymbol[]` and flat `SymbolInformation[]` response shapes (flat is converted and position-sorted); advertises `hierarchicalDocumentSymbolSupport`.
- **Navigation model**: arrowing through the tree reveals the symbol in the editor without stealing focus; enter/click on a leaf jumps and hands focus to the editor; the outline selection follows the editor cursor (nearest enclosing symbol), so it always shows where you are.
- **Built-in fallbacks** when no server can answer: markdown ATX headings (fence-aware, nested by level) and a `go/parser`-based Go outline (functions, methods with receivers, structs/interfaces with members, consts/vars). Covers vendored files and module-less files where gopls answers `no views`, and machines without gopls entirely. The fallback renders immediately as a provisional outline and is replaced by the server response when it arrives.
- **Diagnostic empty states** instead of a silent blank panel: "<binary> is not installed", "No language server for <lang>", "Loading symbols…", and the actual server error text on failure.
- **didOpen fix**: files opened as CLI args only announced the active tab to the server; tabs that become active without passing through `OnFileOpen` now get `didOpen` sent (fixes clangd `-32602: trying to get AST for non-added document` on those tabs, and heals hover/completion/diagnostics for them too). `documentSymbol` retries once to absorb server cold starts.
- **Single-file launches** (`ttt file.go`) now root the language server at the file's directory instead of initializing with an empty `rootURI`.
- Refresh triggers: panel activation, tab switch, save, external file reload. Command palette `Show Outline`, View menu entry. No keybinding (`ctrl+k o` is taken); palette/menu only.

## Testing

- Unit: protocol parsing (both response shapes), markdown heading parser (nesting, fences, edge cases), Go fallback parser (kinds, receivers, partial-parse tolerance), nearest-symbol selection.
- E2E (`tests/e2e/outline_test.go`): fallback rendering, keyboard navigation + jump + focus handoff, click jump, empty state, cursor-follow.
- Functional (`tests/functional/outline.test.js`): headings render, activate jumps and focuses (verified by typing a marker), nesting indentation, empty state.
- Manually verified against gopls (hierarchical), clangd (multi-tab didOpen case), marksman-missing markdown fallback, vendored/orphaned Go files, and a real interactive PTY session.

### CODE REVIEW

File: internal/app/app_lsp.go [hunk 1]
```diff
@@ -594,6 +595,109 @@ func (a *App) ShowReferences(locs []lsp.Location) {
...
+			return
+		}
+	}
+	// Show the built-in outline immediately when available; the server
+	// response replaces it. Otherwise indicate that a request is running.
+	if len(fallback) > 0 {
+		a.ApplySymbols(fallback)
+	} else if a.Symbols.Tree.ItemCount() == 0 {
+		a.Symbols.SetStatus("Loading symbols…")
+	}
```
[the remark this answers is not in the document]
Comment (tenox7) [reply internal/app/app_lsp.go:653]: Addressed in cffcdf2: the loading state now replaces the outline whenever the request targets a different file than the displayed content; same-file refreshes keep the current outline to avoid flicker.

File: internal/app/symbols_panel.go [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+// when no language server is available. Headings nest by level.
+func markdownSymbols(lines []string) []lsp.DocumentSymbol {
+	var headings []mdHeading
+	inFence := false
+	for i, line := range lines {
+		trimmed := strings.TrimSpace(line)
+		if strings.HasPrefix(trimmed, "```") || strings.HasPrefix(trimmed, "~~~") {
+			inFence = !inFence
+			continue
+		}
```
[the remark this answers is not in the document]
Comment (tenox7) [reply internal/app/symbols_panel.go:206]: Addressed in cffcdf2: fences now track their opening marker character, so ~~~ no longer closes a ``` block; added a regression test.

**Review by eugenioenko — APPROVED** [review]
@tenox7 
🥇🥇🥇


Alright, my hat to you! This is an amazing feature flawlessly implemented :)

Thank you so much for your work here and helping TTT become a great IDE, really appreciate it :)  


As a follow up, if you ever get a chance or time, one great addition here would be a search input (like global search) that allows fuzzy searching the symbols within the tree. 


This is gonna land with 1.0.0 release. I'm thinking releasing in a week or two. Want to give a few weeks of codefreeze to verify properly the plugin/widget system

File: internal/app/symbols_panel.go [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+	// OnReveal fires while navigating the tree (selection change); it moves
+	// the editor cursor without taking focus. OnJump fires on activate
+	// (enter/click on a leaf) and also hands focus to the editor.
+	OnReveal func(line, col int)
+	OnJump   func(line, col int)
+}
+
+func NewSymbolsPanel() *SymbolsPanel {
+	sp := &SymbolsPanel{}
+	// Indent 3 keeps leaf labels visually nested past their parent's label
```
Comment (eugenioenko) [inline internal/app/symbols_panel.go:31]: 🎉 
Masterclass! 

I was wondering why indent was not 2 and here it's with  perfect explanation. thank you!
